server/src/session.ts

```diff
@@ -131,6 +131,24 @@ export class Session {
     // project as dirty to force update the graph.
     project.markAsDirty();
     this.info(`Enabling Ivy language service for ${project.projectName}.`);
+
+    // Send diagnostics since we skipped this step when opening the file
+    // (because language service was disabled while waiting for ngcc).
+    this.runGlobalAnalysisForNewlyLoadedProject(project);
+    project.refreshDiagnostics();  // Show initial diagnostics
+  }
+
+  /**
+   * Invoke the compiler for the first time so that external templates get
+   * matched to the project they belong to.
+   */
+  private runGlobalAnalysisForNewlyLoadedProject(project: ts.server.Project) {
+    if (!project.hasRoots()) {
+      return;
+    }
```

Comment:
Can you add a comment that this is case means we have to wait until a TS file manually opened, but is very rare, as you did before?

Reply:
Could you please elaborate more?
In the previous comment, I meant "a project without any root files" is extremely rare. It could be that either the project indeed does not have any files, or something went wrong in the loading phase such that the project is still empty. The second case should not happen at all, and in the first case, there isn't any file that the user could open that'd affect this project.

Reply:
@ayazhafiz could you please take another look and let me know if there's further clarification needed?

Reply:
No more clarification needed, LGTM :)